### PR TITLE
I#100: show account_balance for a user in team/get

### PR DESCRIPTION
**Pull request by shreyansjain**
#100

### CODE REVIEW

**Comment by shreyansjain** [comment]
Wiki has been updated

File: api-team/functions/get/index.js [hunk 1]
```diff
@@ -37,11 +37,13 @@ module.exports.handler = (event, context, callback) => {
             const user_query =
                 SQL `
                 SELECT u.id, u.family_name, u.given_name, u.profile_picture_link,
```
Comment (anmolg) [inline api-team/functions/get/index.js]: Does this query work?

Comment (anmolg) [inline api-team/functions/get/index.js]: Something seems wrong with the way group by is working

File: api-team/functions/get/index.js [hunk 2]
```diff
@@ -37,11 +37,13 @@ module.exports.handler = (event, context, callback) => {
...
-                       r.description as role
+                       r.description as role, sum(tf.amount * tf.quantity) as account_balance
```
Comment (anmolg) [inline api-team/functions/get/index.js]: no need to multiply quantity again?

File: interfaces/user/user.js [hunk 3]
```diff
@@ -31,10 +32,11 @@ User.prototype = {
     toJSON: function () {
         let result = {
             'id': this._id,
-            'name':  this.fullName(),
+            'name': this.fullName(),
             'email': this._email,
             'phone_number': this._phone_number,
-            'profile_picture_link': this._profile_picture_link
+            'profile_picture_link': this._profile_picture_link,
+            'account_balance': this._account_balance || 0
```
Comment (anmolg) [inline interfaces/user/user.js]: any reason why it defaults to 0?

File: api-team/functions/get/index.js (hunk 2 above)
Comment (shreyansjain) [inline api-team/functions/get/index.js]: Removed quantity as it is not used anymore

File: api-team/functions/get/index.js (hunk 1 above)
Comment (shreyansjain) [inline api-team/functions/get/index.js]: yep, seems to be working

File: interfaces/user/user.js (hunk 3 above)
Comment (shreyansjain) [inline interfaces/user/user.js]: otherwise it'll be null. since, on the client side, you want to show the value, null does not make sense, therefore 0

Comment (anmolg) [inline interfaces/user/user.js]: so I think when you sum, if it is null, it should default to 0 there at query level. 

Also, when you return user object, account-balance will always be returned

Comment (shreyansjain) [inline interfaces/user/user.js]: nope, it is default to empty for sum.

we want to  show account_balance irrespective 0 or not 0 value, right?

Comment (anmolg) [inline interfaces/user/user.js]: nope, I think we need to show it when we are returning account balance. But if we dno't need it, we shouldn't default to 0. Unnecessary payload coming back/from. 

```
coalesce(sum(xxx), 0) 
```

Comment (anmolg) [inline interfaces/user/user.js]: Let's set it to 0 at sql level

Comment (shreyansjain) [inline interfaces/user/user.js]: i saw this command but didn't know if we really needed to use it. will add this now

**Comment by shreyansjain** [comment]
Pushed the changes

**Pull request merged**